diff --git a/docroot/core/modules/file/src/Plugin/Field/FieldWidget/FileWidget.php b/docroot/core/modules/file/src/Plugin/Field/FieldWidget/FileWidget.php
index b2386b4..d035997 100644
--- a/docroot/core/modules/file/src/Plugin/Field/FieldWidget/FileWidget.php
+++ b/docroot/core/modules/file/src/Plugin/Field/FieldWidget/FileWidget.php
@@ -394,7 +394,7 @@ public static function process($element, FormStateInterface $form_state, $form)
       $config = \Drupal::config('file.settings');
       $element['description'] = array(
         '#type' => $config->get('description.type'),
-        '#title' => t('Description'),
+        '#title' => t('File Title'),
         '#value' => isset($item['description']) ? $item['description'] : '',
         '#maxlength' => $config->get('description.length'),
         '#description' => t('The description may be used as the label of the link to the file.'),
