diff --git a/docroot/modules/contrib/simple_responsive_table/css/simple-responsive-table.css b/docroot/modules/contrib/simple_responsive_table/css/simple-responsive-table.css
index 2688606..7751446 100644
--- a/docroot/modules/contrib/simple_responsive_table/css/simple-responsive-table.css
+++ b/docroot/modules/contrib/simple_responsive_table/css/simple-responsive-table.css
@@ -33,7 +33,7 @@ td {
 td:before {
   /* Now like a table header */
   position: absolute;
-  left: 1%;
+  left: 14px;
   padding-right: 10px;
   white-space: nowrap;
   text-align: left;
diff --git a/docroot/modules/contrib/simple_responsive_table/js/simple-responsive-table.js b/docroot/modules/contrib/simple_responsive_table/js/simple-responsive-table.js
index e286556..4a51c9a 100644
--- a/docroot/modules/contrib/simple_responsive_table/js/simple-responsive-table.js
+++ b/docroot/modules/contrib/simple_responsive_table/js/simple-responsive-table.js
@@ -21,11 +21,17 @@
         $(context).find('table').once('processTable').each(function () {
           headers = [];
           $(this).find('tr th').each(function () {
-            header = $(this).text();
-            // Handle sort headers.
-            header = header.replace('Sort descending', '');
-            header = header.replace('Sort ascending', '');
-            headers.push(header.trim());
+            if (typeof $(this).data('title') != 'undefined') {
+              header = $(this).data('title').split(',');
+              headers = headers.concat(header);
+            }
+            else {
+              header = $(this).text();
+              // Handle sort headers.
+              header = header.replace('Sort descending', '');
+              header = header.replace('Sort ascending', '');
+              headers.push(header.trim());
+            }
           });
           $(this).find('tr').each(function () {
             $(this).find('td').each(function (index) {
