diff --git a/src/Plugin/Block/TagCloudBlock.php b/src/Plugin/Block/TagCloudBlock.php
index b2c1f7b..36637c7 100644
--- a/src/Plugin/Block/TagCloudBlock.php
+++ b/src/Plugin/Block/TagCloudBlock.php
@@ -127,6 +127,7 @@ class TagCloudBlock extends BlockBase implements ContainerFactoryPluginInterface
         $terms[$term->id()] = [
           'name' => $term->getName(),
           'url' => $url,
+          'type' => $vid,
         ];
       }
     }
diff --git a/templates/default-tag-clouds.html.twig b/templates/default-tag-clouds.html.twig
index 8b1105a..9f4d5af 100644
--- a/templates/default-tag-clouds.html.twig
+++ b/templates/default-tag-clouds.html.twig
@@ -1,7 +1,36 @@
+<div class="row">
+    <div class="col-md-4 col-lg-3">
+        <button type="button" class="btn btn-info file-search-button" id="file-meta-tag-search">Meta Tag Search</button>
+    </div>
+    <div class="col-md-4 col-lg-3">
+        <button type="button" class="btn btn-info file-search-button" id="file-advanced-search">Advanced Search</button>
+    </div>
+    <div class="col-md-3 col-lg-2">
+        <a href="/file-search">
+            <button type="button" class="btn btn-info file-search-button" id="clear-file-search" style="padding: 0 1em;">Clear all filters</button>
+        </a>
+    </div>
+</div>
 <ul class="default_tag_clouds">
-  {% for term_id, term in tags %}
-    <li class="tag_clouds_term">
-      <a href="{{ term.url }}">{{ term.name }}</a>
+    <li class="tag_cloud_term">
+        <a href="/file-search">ALL</a>
     </li>
-  {% endfor %}
+    {% for term_id, term in tags %}
+        {% if (term.type == 'bu') %}
+            <li class="tag_clouds_term term-bu">
+                <a data-tid = "{{ term_id }}" href="{{ term.url }}">{{ term.name }}</a>
+            </li>
+        {% endif %}
+        {% if (term.type == 'division') %}
+            <li class="tag_clouds_term term-division">
+                <a data-tid = "{{ term_id }}" href="{{ term.url }}">{{ term.name }}</a>
+            </li>
+        {% endif %}
+        {% if (term.type == 'language') %}
+            <li class="tag_clouds_term term-language">
+                <a data-tid = "{{ term_id }}" href="{{ term.url }}">{{ term.name }}</a>
+            </li>
+        {% endif %}
+    {% endfor %}
 </ul>
+{{ attach_library('extranet/file-search') }}
